ern-core/src/android.ts

```diff
@@ -290,7 +308,7 @@ export async function installAndLaunchApp({
 
 // Utility method that basically completes whenever the android device is ready
 // It check device readiness every 2 sec (poll way)
-export async function waitForAndroidDevice() {
+export async function waitForAndroidDevice(): Promise<void> {
```

Comment:
I'm a JS/TS noobie, but doesn't a function declared as `async` automatically wrap the return inside of a Promise? Is there a reason you explicitly added the return type here?

Reply:
Yeah, we don't have strong coding style guidelines for this one, and nothing enforced by our linter regarding function return types declaration. Given that, if not enforced through clear coding style guide (and potentially linter), this is subject to individual appreciation. You will probably find some of these in the code base, whereas in other locations we don't explicitly state the return type.

One could say that we must explicitly declare all return types so that it makes it clear to developers as to what the function is returning if they are browsing the code from an IDE that don't have TS integration or just from a text editor. One could say that we need to explicitly declare all return types as long as they are not `void` / `Promise<void>`. One could say that we don't explicitly declare return types no matter what to reduce verbosity at the cost of clarity.

So yes, I'll leave this one to the appreciation of @krunalsshah for now until we eventually decide of stronger coding standards that we ideally enforce automatically through linter and apply it throughout the code base.

Reply:
Agreed. My main point and question is: Why add it to this particular function? Nothing else in the function signature or body changed. I'm aware the linter does not currently check this, but we should try to be consistent at least on a file level. If this is added here (I don't have a strong preference for or against it at this point), then maybe it should also be added to all other functions in the file? Otherwise there's no reason to modify this line.

Reply:
reverted the TS changes.

Reply:
Thanks @krunalsshah :+1: